tools/mti-utils/process_intensities.xml

```diff
@@ -0,0 +1,98 @@
+<tool id="cell_intensity_processing" name="Process single-cell intensities" version="0.0.1">
+    <description>Options to correct for expsure time, autofluorescence subtraction, or compute signal-to-background ratio.</description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="requirements" />
+    <command detect_errors="aggressive"><![CDATA[
+        
+        ln -s '$quant_table' ./quant.csv &&
+
+        ln -s '$channel_csv' ./channels.csv &&
+
+        python '$script'
+
+    ]]></command>
+    <configfiles>
+        <configfile name = "script">
+import os
+import numpy as np
+import pandas as pd
+
+cwd = os.getcwd()
+quant = pd.read_csv(os.path.join(cwd, 'quant.csv'), index_col = 0)
+marker_df = pd.read_csv(os.path.join(cwd, 'channels.csv'))
+
+markers_to_normalize = marker_df['marker_name'].to_list()
+
+for marker in markers_to_normalize:
+
+#if $exp.correct_exposure:
+    exp_time = marker_df.loc[marker_df['marker_name'] == marker, '${exp.exp_col}'].values[0]
+    quant[marker] = quant[marker] / exp_time
+#end if
+
+#if $AF_method.select_method == 'subtract':
+    current_AF_channel = marker_df.loc[marker_df['marker_name'] == marker, '${AF_method.AF_col}'].values[0]
+    if current_AF_channel in markers_to_normalize:
+        quant[marker] = quant[marker] - quant[current_AF_channel]
+        quant[marker] = np.where(quant[marker] &lt; 0, 0, quant[marker])
+#elif $AF_method.select_method == 'SBR':
+    current_AF_channel = marker_df.loc[marker_df['marker_name'] == marker, '${AF_method.AF_col}'].values[0]
+    if current_AF_channel in markers_to_normalize:
+        quant[marker] = quant[marker] / quant[current_AF_channel]
+#end if
+
+quant.to_csv(os.path.join(cwd, 'processed_quant.csv'))
+        </configfile>
+    </configfiles>
+    <inputs>
+        <param name="quant_table" type="data" format="csv" label="Input quantification table (csv)" />
```

Comment:
Hi @bgruening, thank you for your comments! Our multiplex imaging tools are quite heavily invested in CSV files both as inputs and outputs, and much of this stems from external tool developers so it would be a challenge to change over to tabular. Other tools in the workflow use the same comma-separated channel metadata file as this tool does, so we'd like to keep that consistent across tools if we can. With this, I have a couple questions if you don't mind: 

- Does the data_column param work with CSV files? In my experimenting with it, it doesn't seem to be able to parse the header names when the input file is CSV (using  `use_header_names="true"`)

- Why are tabular files more supported in Galaxy? Is that a purposeful choice or just how it has ended up by coincidence? 

Thank you for your time!

Reply:
For your first question, I do not know to be honest. Since CSV is not very popular in Galaxy there can be bugs or not implemented features.
For your second question, there are so many variants of a CSV that's it was hard I think to support all. In addition if you don't want to increase the complexity of all text processing tools significantly you need to choose between tsv and CSV. And tsv is what most of our tools support oob. You just get more tools, better Vis and in general better integration at this point.
It can change of course, but their is some benefit for the Galaxy community to stick to one standard and abstract the complexity away on the tool level. By e.g. taking an CSV input but converting it and only pass out tsv.

Reply:
Hi @bgruening thanks for the response! For the time being, we'd like to stick with CSVs because this keeps us in sync with the Nextflow implementation of MCMICRO, which is important for consistency and reproducibility of our MTI workflows, and allows us to easily incorporate any changes or expansions that they make. Since the `data_column` data type does not work with CSVs, we have stuck with keeping the column selection parameter as a `text` type. We really appreciate your reviews, and have opened an issue reflecting the suggested changes to tabular and `data_column` here: https://github.com/goeckslab/tools-mti/issues/44 

All of your other comments have been resolved. Thanks again!